### PR TITLE
Covertiy fixes

### PR DESCRIPTION
**Pull request by ifranzki**
Check errors for all RW-lock or mutex initialization calls, and perform proper error handling and cleanup if it fails.

### CODE REVIEW

File: usr/lib/common/lock_btree.c [hunk 1]
```diff
@@ -502,7 +502,18 @@ void bt_init(struct btree *t, void (*delete_func)(void *))
...
      */
-    pthread_mutexattr_init(&attr);
-    pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE);
-    pthread_mutex_init(&t->mutex, &attr);
+    if (pthread_mutexattr_init(&attr) != 0) {
+        TRACE_ERROR("Initializing mutex attribute failed.\n");
+        return CKR_CANT_LOCK;
+    }
+    if (pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE) != 0) {
+        TRACE_ERROR("Initializing mutex attribute failed.\n");
```
Comment (jschmidb) [inline usr/lib/common/lock_btree.c:510]: Although the trace will show the source code line numbers, we could use different trace messages for these 3 possible errors. Like 
TRACE_ERROR("pthread_mutexattr_init failed.\n");
TRACE_ERROR("pthread_mutexattr_settype failed.\n");
TRACE_ERROR("pthread_mutex_init failed.\n");

File: usr/lib/icsf_stdll/icsf_specific.c [hunk 2]
```diff
@@ -375,8 +375,17 @@ CK_RV icsftok_init(STDLL_TokData_t * tokdata, CK_SLOT_ID slot_id,
...
         return CKR_HOST_MEMORY;
     list_init(&icsf_data->sessions);
-    pthread_mutex_init(&icsf_data->sess_list_mutex, NULL);
-    bt_init(&icsf_data->objects, free);
+    if (pthread_mutex_init(&icsf_data->sess_list_mutex, NULL) != 0) {
+        TRACE_ERROR("Initializing session list lock failed.\n");
+        free(icsf_data);
+        return CKR_CANT_LOCK;
+    }
+   if (bt_init(&icsf_data->objects, free) != CKR_OK) {
```
Comment (jschmidb) [inline usr/lib/icsf_stdll/icsf_specific.c:383]: Indentation?

File: usr/lib/common/lock_btree.c (hunk 1 above)
Comment (ifranzki) [reply usr/lib/common/lock_btree.c:510]: sure, it was actually my intention to use 3 different messages, but somehow I forgot to change them.

File: usr/lib/icsf_stdll/icsf_specific.c (hunk 2 above)
Comment (ifranzki) [reply usr/lib/icsf_stdll/icsf_specific.c:383]: fixed

**Review by jschmidb — APPROVED** [review]
lgtm

**Pull request merged**